Integrations/MimecastV2/MimecastV2.py

```diff
@@ -1479,6 +1471,605 @@ def download_attachment_request(attachment_id):
+    entry_context = find_groups_api_response_to_context(api_response)
+
+    return_outputs(markdown_output, entry_context, api_response)
+
+
+def create_find_groups_request():
+    api_endpoint = '/api/directory/find-groups'
+    query_string = demisto.args().get('query_string', '').encode('utf-8')
+    query_source = demisto.args().get('query_source', '').encode('utf-8')
+    pagination = demisto.args().get('pagination')
+
+    meta = dict()  # type: Dict[str, Dict[str, int]]
+    data = dict()  # type: Dict[str, Dict[str, str]]
+
+    if pagination:
+        meta['pagination'] = {
+            'pageSize': int(pagination)
+        }
+
+    if query_string:
+        data['query'] = query_string
+    if query_source:
+        data['source'] = query_source
+
+    payload = {
+        'meta': meta,
+        'data': [data]
+    }
+
+    response = http_request('POST', api_endpoint, str(payload))
+    if isinstance(response, dict) and response.get('fail'):
+        return_error(json.dumps(response.get('fail', [{}])[0].get('errors')))
+    return response
+
+
+def find_groups_api_response_to_markdown(api_response):
+    num_groups_found = api_response.get('meta', {}).get('pagination', {}).get('pageSize', 0)
+    query_string = demisto.args().get('query_string', '')
+    query_source = demisto.args().get('query_source', '')
+
+    if not num_groups_found:
+        md = '### Found 0 groups'
+
+        if query_string:
+            md += '\n#### query: ' + query_string
+
+        if query_source:
+            md += '\n#### source: ' + query_source
+
+        return md
+
+    md = 'Found ' + str(num_groups_found) + ' groups:'
+
+    if query_string:
+        md += '\n#### query: ' + query_string
+
+    if query_source:
+        md += '\n#### source: ' + query_source
+
+    groups_list = list()
+    for group in api_response.get('data', [])[0]['folders']:
+        group_entry = {
+            'Name': group['description'],
+            'Source': group['source'],
+            'Group ID': group['id'],
+            'Number of users': group['userCount'],
+            'Parent ID': group['parentId'],
+            'Number of child groups': group['folderCount']
+        }
+
+        groups_list.append(group_entry)
+
+    md = tableToMarkdown(md, groups_list,
+                         ['Name', 'Source', 'Group ID', 'Number of users', 'Parent ID', 'Number of child groups'])
+
+    return md
+
+
+def find_groups_api_response_to_context(api_response):
+    groups_list = list()
+    for group in api_response['data'][0]['folders']:
+        group_entry = {
+            'Name': group['description'],
+            'Source': group['source'],
+            'ID': group['id'],
+            'NumberOfUsers': group['userCount'],
+            'ParentID': group['parentId'],
+            'NumberOfChildGroups': group['folderCount']
+        }
+
+        groups_list.append(group_entry)
+
+    return {'Mimecast.Groups(val.ID && val.ID == obj.ID)': groups_list}
```

Comment:
We don't like those things in plural because we might have only one results in the Context
```suggestion
    return {'Mimecast.Group(val.ID && val.ID == obj.ID)': groups_list}
```

Reply:
Done.